### ISSUE TITLE
`git lab` creates new MRs instead of updating existing MRs

### ISSUE THREAD

**Issue by kostajh** [yaoyuannnn/gitlab-gerrit-review#49]
Steps to reproduce:

- `glab mr checkout 98` (https://gitlab.wikimedia.org/repos/mediawiki/services/ipoid/-/merge_requests/98)
- Make a change, `git commit`, rebase interactive and update the commit associated with MR 96 (https://gitlab.wikimedia.org/repos/mediawiki/services/ipoid/-/merge_requests/96)
- `git lab`

Expected:

- MRs 97 and 98 are unchanged, MR 96 has an updated commit message and MR description

Actual:

- New MRs are opened for all three commits.

**Comment by kostajh** [comment]
With the patch in #50 (and a few `print()` statements), I see this: 

```
Commits to be reviewed:
* 90162e9c shell: Make scripts portable for macOS

Proceed? ([y]/n) y
{'macos-6b8f': <gerritlab.merge_request.MergeRequest object at 0x101d246a0>}
macos-6b8f-6b8f

Proposed MRs:

No existing MRs will be updated
MRs to create
* (new) shell: Make scripts portable for macOS
    macos-6b8f-6b8f -> main
```

The branch I'm on is `macos-6b8f`. That's als what I see at https://gitlab.wikimedia.org/repos/mediawiki/services/ipoid/-/merge_requests/122. But when running `git lab`, it looks like the `source_branch` property is incorrectly identified as `macos-6b8f-6b8f`, instead of `macos-6b8f`

**Comment by kostajh** [comment]
> With the patch in #50 (and a few `print()` statements), I see this:
> 
> ```
> Commits to be reviewed:
> * 90162e9c shell: Make scripts portable for macOS
> 
> Proceed? ([y]/n) y
> {'macos-6b8f': <gerritlab.merge_request.MergeRequest object at 0x101d246a0>}
> macos-6b8f-6b8f
> 
> Proposed MRs:
> 
> No existing MRs will be updated
> MRs to create
> * (new) shell: Make scripts portable for macOS
>     macos-6b8f-6b8f -> main
> ```
> 
> The branch I'm on is `macos-6b8f`. That's als what I see at https://gitlab.wikimedia.org/repos/mediawiki/services/ipoid/-/merge_requests/122. But when running `git lab`, it looks like the `source_branch` property is incorrectly identified as `macos-6b8f-6b8f`, instead of `macos-6b8f`

Maybe branch name is the wrong thing to check, though. We have `Change-Id` which should be the source of proof for saying "Commit X belongs in MR Y". Should we switch to using that, rather than relying on branch name?

**Comment by dancysoft** [comment]
I think this is probably resolved as of #54.  Note that upgrading to the latest Gerritlab code means that existing MRs with the old naming scheme will de-associated.
